src/Adapter/Image/ImageRetriever.php

```diff
@@ -190,6 +199,48 @@ public function getImage($object, $id_image)
                 );
             }
 
+            $generateAdditionalWebP = (bool) Configuration::get('PS_ADDITIONAL_IMAGE_WEBP');
+
+            if ($generateAdditionalWebP) {
+                $resizedImagePathWebP = implode(DIRECTORY_SEPARATOR, [
+                    $imageFolderPath,
+                    $id_image . '-' . $image_type['name'] . '.webp',
+                ]);
+
+                if (!file_exists($resizedImagePathWebP)) {
+                    ImageManager::resize(
+                        $originalImagePath,
+                        $thumbnailPath,
+                        (int) $image_type['width'],
+                        (int) $image_type['height'],
+                        'webp',
+                        true
+                    );
+                }
+
+                $additionalSources['webp'] = $this->link->$getImageURL($rewriteLink, $id_image, $image_type['name'], '.webp');
```

Comment:
```suggestion
                $additionalSources['webp'] = $this->link->getImageURL($rewriteLink, $id_image, $image_type['name'], '.webp');
```

I think it's a type, right?

Reply:
good catch

Reply:
no, the variable is initialized before.